my-project/lib/bowlingScore.js

```diff
@@ -0,0 +1,77 @@
+class BowlingScore {
+  constructor(scoresheet) {
+    this.scoresheet = scoresheet;
+    this.nineFrames = this.scoresheet.slice(0, 9);
+    this.total = [];
+    this.sparePoints = 0;
+    this.strikePoints = 0;
+  }
+
+  frame() {
+    // calculate points for frames that are not strikes or spares
+    const notBonusFrames = this.scoresheet.slice(0, 10);
+    // check if a frame sum is less than 10
+    // if yes then add the total
+    // Then push the total into total array
+    notBonusFrames.forEach((frame) => {
+      const roll = frame.reduce((a, b) => a + b, 0);
+      if (roll < 10) {
+        this.total.push(frame);
+      }
+    });
+  }
+
+  spares() {
+    // check if sum of each frame is 10 and if the first roll does not equal 10
+    this.nineFrames.forEach((frame) => {
+      let index = this.nineFrames.indexOf(frame);
+      let nextFrame = this.scoresheet[index + 1];
```

Comment:
DRY:

Not sure if this works:
array = return of a function as an array [index , nextFrame]